draft-ietf-scitt-architecture.md

```diff
@@ -552,64 +581,131 @@ Editor's Note: This may be moved to appendix.
 POST <Base URL>/entries
 ~~~
 
+Headers:
+
+- `Content-Type: application/cose`
+
 Body: SCITT COSE_Sign1 message
 
 #### Response
 
 One of the following:
 
-- HTTP Status `201` - Registration was tentatively successful pending service consensus.
-- HTTP Status `400` - Registration was unsuccessful.
-  - Error code `AwaitingDIDResolutionTryLater`
-  - Error code `InvalidInput`
+- Status 201 - Registration is successful.
+  - Header `Location: <Base URL>/entries/<Entry ID>`
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID"> }`
 
-[TODO] Use 5xx for AwaitingDIDResolutionTryLater
+- Status 202 - Registration is running.
+  - Header `Location: <Base URL>/operations/<Operation ID>`
+  - Header `Content-Type: application/json`
+  - (Optional) Header: `Retry-After: <seconds>`
+  - Body `{ "operationId": "<Operation ID>", "status": "running" }`
 
-The `201` response contains the `x-ms-ccf-transaction-id` HTTP header which can be used to retrieve the Registration Receipt with the given transaction ID. [TODO] this has to be made generic
+- Status 400 - Registration was unsuccessful due to invalid input.
+  - Error code `badSignatureAlgorithm`
+  - [TODO]: more error codes to be defined
```

Comment:
Perhaps add a github issue to define more error codes and refer it here!

Reply:
Done.